File: app/src/main/java/jp/ad/wide/sfc/arch/attendancesystem/net/PortalClient.java
package jp.ad.wide.sfc.arch.attendancesystem.net;

import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.toolbox.JsonObjectRequest;

import org.json.JSONException;
import org.json.JSONObject;

import jp.ad.wide.sfc.arch.attendancesystem.AppController;

public class PortalClient {
    final static String API_V1_URL = "https://portal.sfc.wide.ad.jp/api/v1";
    protected String mAccessToken;
    protected Response.Listener<JSONObject> mSuccessListener;
    protected Response.ErrorListener mErrorListener;

    public PortalClient(Response.Listener<JSONObject> successListener, Response.ErrorListener errorListener, String accessToken) {
        mSuccessListener = successListener;
        mErrorListener = errorListener;
        mAccessToken = accessToken;
    }

    public void createAttendance(String studentNumber) {
        JSONObject parameters = new JSONObject();
        try {
            parameters.put("access_token", mAccessToken);
            parameters.put("student_id", studentNumber);
        } catch (JSONException e) {
            e.printStackTrace();
        }

        JsonObjectRequest jsonObjReq = new JsonObjectRequest(
                Request.Method.POST, API_V1_URL + "/attendances", parameters, mSuccessListener, mErrorListener);
        AppController.getInstance().addToRequestQueue(jsonObjReq, "/attendances");
    }
}

File: app/src/main/java/jp/ad/wide/sfc/arch/attendancesystem/net/PortalClient.kt
package jp.ad.wide.sfc.arch.attendancesystem.net

import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import jp.ad.wide.sfc.arch.attendancesystem.AppController
import org.json.JSONObject

class PortalClient(
    private val appController: AppController,
    private val successListener: Response.Listener<JSONObject>,
    private val errorListener: Response.ErrorListener,
    private val accessToken: String
) {

    fun createAttendance(studentNumber: String) {
        val parameters = JSONObject().apply {
            put("access_token", accessToken)
            put("student_id", studentNumber)
        }

        val jsonObjReq = JsonObjectRequest(
            Request.Method.POST,
            "$API_V1_URL/attendances",
            parameters,
            successListener,
            errorListener
        )
        appController.addToRequestQueue(jsonObjReq, "/attendances")
    }

    companion object {
        private const val API_V1_URL = "https://portal.sfc.wide.ad.jp/api/v1"
    }
}
